spec/puppetserver/ca/cli_spec.rb

```diff
@@ -335,4 +342,43 @@
       end
     end
   end
+
+  describe 'the generate action' do
+    let(:usage) { /.*Usage:.* puppetserver ca generate.*Display this generate specific help output.*/m }
+
+    include_examples 'basic cli args',
+      'generate',
+      /.*Usage:.* puppetserver ca generate.*Display this generate specific help output.*/m
+
+    it 'prints the help output & returns 1 if invalid flags are given' do
+      exit_code = Puppetserver::Ca::Cli.run(['generate', '--hello'], stdout, stderr)
+      expect(stderr.string).to match(/Error.*--hello/m)
+      expect(stderr.string).to match(usage)
+      expect(exit_code).to eq(1)
+    end
+
+    it 'does not print the help output if called correctly' do
+      Dir.mktmpdir do |tmpdir|
+        with_temp_cadir tmpdir do |conf|
+          exit_code = Puppetserver::Ca::Cli.run(['generate', '--config', conf], stdout, stderr)
+          expect(stderr.string).to be_empty
+          expect(stdout.string.strip).to eq("Generation succeeded. Find your files in #{tmpdir}/ca")
```

Comment:
^^ I had to add .strip to get this to work.  I guess thats expected for stdout.string to add the space at the end??  I looked at the output and didnt see any issues.

Reply:
It probably adds a newline.